python/semantic_kernel/template_engine/blocks/code_block.py

```diff
@@ -104,45 +107,29 @@ def _enrich_function_arguments(
     ) -> "KernelArguments":
         function_block: FunctionIdBlock = self.tokens[0]
-        # current templates only support 1 argument.
-        function_argument: Block = self.tokens[1]
+        first_function_argument: Block = self.tokens[1]
 
         if not function_metadata.parameters:
             raise ValueError(
                 f"Function {function_block.plugin_name}.{function_block.function_name} does not take any arguments "
                 f"but it is being called in the template with {len(self.tokens) - 1} arguments."
             )
-
-        # if function_argument.type != BlockTypes.NAMED_ARG:
-        logger.debug(f"Passing variable/value: `{self.tokens[1].content}`")
-        rendered_value = function_argument.render(kernel, arguments)
-        if function_argument.type == BlockTypes.VALUE:
-            arguments[function_metadata.parameters[0].name] = rendered_value
-        elif function_argument.type == BlockTypes.VARIABLE:
-            arguments[function_argument.name] = rendered_value
-        else:
-            raise ValueError("Unknown block type: %s", self.tokens[0].type)
-            # first_positional_parameter_name = function_metadata.parameters[0].name
-            # first_positional_input_value = function_argument.render(kernel, arguments)
-            # arguments[first_positional_parameter_name] = first_positional_input_value
-            # named_args_start_index += 1
-
-        # Commented out because not needed until ADR0009 is implemented
-        # for i in range(named_args_start_index, len(self.tokens)):
-        #     arg = self.tokens[i]
-        #     if arg.type != BlockTypes.NAMED_ARG:
-        #         error_msg = "Functions support up to one positional argument"
-        #         logger.error(error_msg)
-        #         raise Exception(f"Unexpected first token type: {arg.type}")
-
-        #     if first_positional_parameter_name and first_positional_parameter_name.lower() == arg.name.lower():
-        #         raise ValueError(
-        #             f"Ambiguity found as a named parameter '{arg.name}' cannot be set for the first parameter "
-        #             f"when there is also a positional value: '{first_positional_input_value}' provided. "
-        #             f"Function: {function_block.plugin_name}.{function_block.function_name}"
-        #         )
-
-        #     arguments[arg.name] = arg.get_value(arguments)
+        named_args_start_index = 1
+
+        if first_function_argument.type != BlockTypes.NAMED_ARG:
+            logger.debug(f"Passing variable/value: `{self.tokens[1].content}`")
+            rendered_value = first_function_argument.render(kernel, arguments)
+            first_positional_parameter_name = function_metadata.parameters[0].name
+            arguments[first_positional_parameter_name] = rendered_value
+            named_args_start_index += 1
+
+        for i in range(named_args_start_index, len(self.tokens)):
```

Comment:
@juliomenendez for this code, we need to be able to loop over a range of tokens that is a subset of the overall tokens. Using enumerate would have us loop through all tokens from 0 to n, right? The `range(named_args_start_index, len(self.tokens))` allows us to start at index 1 and loop until `len(self.tokens)`.

Reply:
No no, you can specify a starting point, in this case it would look like this:
```python
for i, token in enumerate(self.tokens, start=named_args_start_index):
```

Reply:
Thanks for the comments, I've redone the logic more generically to make it easier to read.